### ISSUE TITLE
Should not need '!' for accessing param fields and similar

### ISSUE THREAD

**Issue by vasslitvinov** [chapel-lang/chapel#13617]
When accessing a param field of a variable of a nilable type, there is no need for `!`. The same holds in a couple of other generic cases.

```chpl
module w {

  class C1 { param n; type t; }
  var c1: owned C1(5,real)?;
  writeln(c1.n);          // since #14549, this no longer requires '!' after 'c1'
  writeln(c1.t:string);   // ditto

  class C2 { var x; }
  var c2: owned C2(int)?;
  writeln(c2.x.type:string);    // currently this requires '!' after 'c2'

}
```

This is a convenience feature, as these should be compile-time operations so should not involve any runtime checking.

Example tests:
```
# see proc getSuperType()
test/classes/diten/nearestMutualParentClass.chpl
```

**Comment by mppf** [comment]
Makes sense to me.

**Comment by vasslitvinov** [comment]
Resolved by #14549.
